src/algorithms/digi/PulseNoise.cc

```diff
@@ -27,11 +27,32 @@ void PulseNoise::process(const PulseNoise::Input& input, const PulseNoise::Outpu
     out_pulse.setInterval(pulse.getInterval());
     out_pulse.setTime(pulse.getTime());
 
+    float integral = 0;
     //Add noise to the pulse
     for (int i = 0; i < pulse.getAmplitude().size(); i++) {
       double noise = m_noise(generator) * m_cfg.scale;
       out_pulse.addToAmplitude(pulse.getAmplitude()[i] + noise);
+      integral += pulse.getAmplitude()[i] + noise;
     }
+
+#if EDM4EIC_VERSION_MAJOR >= 8 && EDM4EIC_VERSION_MINOR >= 1
+    out_pulse.setIntegral(integral);
+    out_pulse.setPosition(pulse.getPosition());
+    for (auto subpulse : pulse.getPulses()) {
+      out_pulse.addToPulses(subpulse);
+    }
```

Comment:
I need another opinion on whether this relationship should just include the pulse that noise was added to, or all of the pulses which formed that pulse.

Reply:
I would just point to the pulses from the input collections, but this is fine too.

Reply:
Ok, swapped to that. Should the hits and particle relations both be kept too? The pulse relations is the only one which is really an either or.

Reply:
I would keep those. It doesn't cost us anything to carry them over, and we'd need to add logic to recursively traverse them otherwise.